### PR TITLE
No underscores are allowed in DNSNames before BR 1.6.2's permissibility period

### PR DESCRIPTION
**Pull request by christopher-henderson**
This lint encodes the notion that, before a brief period of permissibility from December 10th 2018 to April 1st 2019, that CABF BRs implicitly adopted the stance of RFC5280 that underscores are not permitted within DNS names.

### CODE REVIEW

File: v3/integration/config.json [hunk 1]
```diff
@@ -770,6 +770,9 @@
     },
     "w_subject_given_name_recommended_max_length": {},
     "w_subject_surname_recommended_max_length": {},
-    "w_tls_server_cert_valid_time_longer_than_397_days": {}
+    "w_tls_server_cert_valid_time_longer_than_397_days": {},
+    "e_no_underscores_before_1_6_2": {
+      "ErrCount": 340
```
Comment (sleevi) [inline v3/integration/config.json:775]: Do you have some examples just to double check (e.g. we should be able to find these in Moz incident reports)

File: v3/lints/cabf_br/lint_no_underscores_before_1_6_2_test.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+			InputFilename:  "dNSNameNoUnderscores.pem",
+			ExpectedResult: lint.Pass,
+		},
+		{
+			Name:           "An underscores",
+			InputFilename:  "dNSNameWithUnderscores.pem",
+			ExpectedResult: lint.Error,
+		},
+		{
+			Name:           "Not effective",
```
Comment (sleevi) [inline v3/lints/cabf_br/lint_no_underscores_before_1_6_2_test.go:41]:
```suggestion
			Name:           "After ineffective date / after Ballot 1.6.2",
```

File: v3/testdata/dNSNameNoUnderscores.pem [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+                    04:89:6d:b1:15:7a:e5:f1:7f:6e:6c:46:d6:2e:0b:
+                    32:9a:c1:72:a8:96:a8:b0:ed:20:1f:49:64:71:2c:
+                    dc:00:f0:7d:90:47:c8:f1:f3:f4:c3:33:dd:f7:af:
+                    98:17:69:73:b2:05:35:74:75:96:4d:66:8a:6f:1a:
+                    58:b6:e0:a5:2c
+                ASN1 OID: prime256v1
+                NIST CURVE: P-256
+        X509v3 extensions:
+            X509v3 Subject Alternative Name: 
+                DNS:no.underscores.com
```
Comment (sleevi) [inline v3/testdata/dNSNameNoUnderscores.pem:24]: If it's not _too_ much hassle, is it possible to put these under the `.test` or `.example` (c.f. https://www.iana.org/domains/reserved )

**Review by sleevi — CHANGES_REQUESTED** [review]

File: v3/integration/config.json (hunk 1 above)
Comment (christopher-henderson) [reply v3/integration/config.json:775]: Ah, certainly. Better than some examples here is the full dataset (first with only a list of fingerprints, and second with all pem data and openssl textual outputs).

[failedFingerprints.txt](https://github.com/zmap/zlint/files/8063337/failedFingerprints.txt)
[failedPems.pem](https://github.com/zmap/zlint/files/8063341/failures.txt)

Here are some smoke checks from crt.sh

https://crt.sh/?q=005155a41690d7ce8ad69556f5b2aafea595298e0985a2a70e7210ac426fd5c4
https://crt.sh/?q=018e2a96f6ad8d517e001afd4a0d5d700d9fd7093e9c8349bd7591139504fcad
https://crt.sh/?q=e6769a8ebd3b9801aa4abb329880fdd1da1bb2002ec070ed97e267ee34bf3337

File: v3/testdata/dNSNameNoUnderscores.pem (hunk 3 above)
Comment (christopher-henderson) [reply v3/testdata/dNSNameNoUnderscores.pem]: No trouble! I swear, half the ideas I come up with for maintaining this repo have to do with easy test data generation.

**Review by sleevi — APPROVED** [review]

**Pull request merged**